### ISSUE TITLE
Is it possible to send a txt file to it or a string via curl in cmd or some such? 

### ISSUE THREAD

**Issue by GamingDaveUk** [daswer123/xtts-api-server#22]
Currently using https://github.com/ParisNeo/lollms-webui and would like to turn some of the generated text into a wav file using our story teller narrator voice like we do with silly tavern. Have requested xtts api server to be supported but wondering if i can send 3 or 4 paragraphs to the xtts server manually and get a wav file created?

**Comment by daswer123** [comment]
Of course there is an endpoint `tts_to_file`

![image](https://github.com/daswer123/xtts-api-server/assets/22278673/c15b4ebd-8a59-4ff5-96f4-8868af4f9716)

**Comment by daswer123** [comment]
Yes approximately like that, you can see more details in the documentation ipOfServer:Port/docs

Good idea , I will soon add the ability to process the path in a text field and load a text file

**Comment by GamingDaveUk** [comment]
Actually a bit stuck with the api doc
I used the test function on it with this in the request body
`{
  "text": "Hello world",
  "speaker_wav": "Dave.wav",
  "language": "English",
  "file_name_or_path": "output.wav"
}`

Dave.wav exists as a wav file 
language English seems right from what i can see in silly tavern
output path i assume is correct.

this puts the curl as:

`curl -X 'POST' \
  'http://localhost:8020/tts_to_file' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "text": "Hello world",
  "speaker_wav": "Dave.wav",
  "language": "English",
  "file_name_or_path": "output.wav"
}'`

request url is http://localhost:8020/tts_to_file

the response is an unhelpful:

500 Undocumented | Error: Internal Server Error

```
{
  "detail": "An error occurred: "
}
```

What am i doing wrong?

**Comment by GamingDaveUk** [comment]
ah language needs to be en not English

**Comment by daswer123** [comment]
The language should be in the form of a language code, the tavern specifically makes it visually so, but actually sends the language code

**Comment by GamingDaveUk** [comment]
Sussed it and with the help of chatgpt I created a bat file i can drop text files on and have them turned into .wav files.

Code for the batfile:
`@echo off
setlocal enabledelayedexpansion

rem Set the API endpoint and function
set API_ENDPOINT=http://localhost:8020/tts_to_file

rem Set the input values
set SPEAKER_WAV="dave2.wav"
set LANGUAGE="en"
set FILE_NAME_OR_PATH="test.wav"

rem Check if a file is dropped onto the batch file
if "%~1" neq "" (
    set "TEXT_FILE=%~1"
) else (
    echo No text file dropped. Exiting.
    exit /b
)

rem Read the contents of the dropped text file into the TEXT variable
set "TEXT="
for /f "delims=" %%i in ('type "%TEXT_FILE%"') do (
    set "LINE=%%i"
    rem Escape special characters in the line
    set "LINE=!LINE:"=\"!"
    set "TEXT=!TEXT!!LINE! "
)

rem Trim trailing whitespace
set "TEXT=!TEXT:~0,-1!"

rem Build the JSON payload
set JSON_PAYLOAD={^
  "text": "!TEXT!",^
  "speaker_wav": %SPEAKER_WAV%,^
  "language": %LANGUAGE%,^
  "file_name_or_path": %FILE_NAME_OR_PATH%^
}

rem Write the JSON payload to a temporary file
echo %JSON_PAYLOAD% > temp.json

rem Make the curl request
curl -v -X POST -H "Content-Type: application/json" -d @temp.json %API_ENDPOINT%

rem Remove the temporary file
del temp.json
`

I Dont know if chatgpt made any mistakes, python is not a language i know, how ever its working well

Cheers for the help and pointing to the docs (wish i had noticed that sooner) 
Now i really should install XTTS seperate from silly tavern lol

**Comment by daswer123** [comment]
Great, glad the issue has been resolved. If you need XTTS separate from the tavern, you can use this interface.

https://github.com/daswer123/xtts-webui

**Comment by GamingDaveUk** [comment]
I was planning to just install the server seperate, i had no idea you had a full blown web ui for it lol.... sort of makes what i did null and void for my purpose lol

**Comment by ParisNeo** [comment]
Thanks alot, this is a very very good TTS service.
I have integrated it into lollms so that any persona on lollms can use it to have its unique voice.
![image](https://github.com/daswer123/xtts-api-server/assets/827993/9334a375-9d9f-4fba-ba5c-b72bbc7f2abb)
@daswer123 Thanks alot for this great work. I have created a LoLLMS wrapper to this service just like I did with Stable diffusion. Ofcourse I give link  to the repository and the licence. As it is MIT, I guess you have no problem for me using it and integrating it into lollms. I always keep the credits to the original creators so there will be no ambiguity about the work.
![image](https://github.com/daswer123/xtts-api-server/assets/827993/f4579b9f-cbfc-4e4e-a374-93759ca84531)

![image](https://github.com/daswer123/xtts-api-server/assets/827993/e6c51ea3-d10e-4d44-aef7-f2b371930d8f)

Have a nice evening
